binding.c

```diff
@@ -514,6 +517,56 @@ static const char* execute_hmac(
   return NULL;
 }
 
+static const char* execute_sign(
+  const int nid,
+  const unsigned char* key,
+  const int key_size,
+  const unsigned char* source,
+  const int source_size,
+  unsigned char* target,
+  int* target_size
+) {
+  RSA *rsa = NULL;
+  BIO *keybio = BIO_new_mem_buf((void*)key, key_size);
+  if (keybio == NULL) {
+    return "key buffer allocation failed";
+  }
+  rsa = PEM_read_bio_RSAPrivateKey(keybio, &rsa, NULL, NULL);
+  BIO_free(keybio);
+  if (rsa == NULL) {
+    return "invalid private key";
+  }
+  EVP_MD_CTX* m_RSASignCtx = EVP_MD_CTX_create();
```

Comment:
Can we stick to the `ctx` naming convention as per the other `execute_` methods?

Reply:
done.